### PR TITLE
BuildResidentialHPXML: make geothermal loop config argument optional

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Make `geothermal_loop_configuration` optional. Not providing a value means no geothermal loop is added.
Also, check for a GSHP when trying to add a geothermal loop. Throw a warning, and skip adding the geothermal loop, if a GSHP is not found.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] ~Schematron validator (`EPvalidator.xml`) has been updated~
- [x] Sample files have been added/updated (`openstudio tasks.rb update_hpxmls`)
- [x] Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests/test*.rb` and/or `workflow/tests/test*.rb`)
- [ ] ~Documentation has been updated~
- [ ] ~Changelog has been updated~
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -1384,10 +1384,9 @@ def arguments(model) # rubocop:disable Lint/UnusedMethodArgument
     # geothermal_loop_configuration_choices << HPXML::GeothermalLoopLoopConfigurationOther
     geothermal_loop_configuration_choices << HPXML::GeothermalLoopLoopConfigurationVertical
 
-    arg = OpenStudio::Measure::OSArgument::makeChoiceArgument('geothermal_loop_configuration', geothermal_loop_configuration_choices, true)
+    arg = OpenStudio::Measure::OSArgument::makeChoiceArgument('geothermal_loop_configuration', geothermal_loop_configuration_choices, false)
```
Comment (joseph-robertson) [inline BuildResidentialHPXML/measure.rb:1387]: Should we just remove the "none" choice altogether? Leaving the choices array with only "vertical" as an option?

Comment (shorowit) [reply BuildResidentialHPXML/measure.rb:1387]: Either way seems fine. But I just noticed that we can create a `GeothermalLoop` in the HPXML file even if the user didn't select a GSHP for their HVAC system. Shouldn't we check that?

File: workflow/hpxml_inputs.json [hunk 2]
```diff
@@ -105,7 +105,6 @@
     "heat_pump_backup_type": "none",
     "heat_pump_backup_fuel": "electricity",
     "heat_pump_backup_heating_efficiency": 0,
-    "geothermal_loop_configuration": "none",
```
Comment (shorowit) [inline workflow/hpxml_inputs.json:108]: 👍

**Review by shorowit — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**